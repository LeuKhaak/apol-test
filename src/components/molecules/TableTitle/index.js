import React from "react";
import Cell from "../Cell";

const TableTitle = ({ el, sortUsers }) => {
  return (
    <div
      id={el[0]}
      key={el[0]}
      style={{ cursor: "pointer" }}
      onClick={() => sortUsers(el[0])}
    >
      <Cell content={el[1]} cellStyle="cellTitle" cellContentStyle="simbol" />
    </div>
  );
};

export default TableTitle;
