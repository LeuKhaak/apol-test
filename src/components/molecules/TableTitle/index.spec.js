import React from "react";
import { shallow } from "enzyme";
import TableTitle from ".";

let component = null;
const onClick = jest.fn();
const setComponent = (props) => shallow(<TableTitle {...props} />);

describe("TableTitle testing:", () => {
  test("is TableTitlerendered", () => {
    component = setComponent({
      el: ["email", "gfdr"],
    });
    expect(component).toMatchSnapshot();
  });

  test("check onClickFunc", () => {
    component = setComponent({
      el: ["email", "gfdr"],
      sortUsers: onClick,
    });
    component.find("div").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
