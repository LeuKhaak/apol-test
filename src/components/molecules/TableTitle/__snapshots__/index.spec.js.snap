// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableTitle testing: is TableTitlerendered 1`] = `
<div
  id="email"
  key="email"
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <Cell
    cellContentStyle="simbol"
    cellStyle="cellTitle"
    content="gfdr"
  />
</div>
`;
